server/authz/authz_test.go

```diff
@@ -0,0 +1,500 @@
+		{user: nil, object: query, action: write, allow: false},
+		{user: nil, object: query, action: run, allow: false},
+		{user: nil, object: observerQuery, action: read, allow: false},
+		{user: nil, object: observerQuery, action: write, allow: false},
+		{user: nil, object: observerQuery, action: run, allow: false},
+
+		// User can still read queries with no roles
+		{user: test.UserNoRoles, object: query, action: read, allow: true},
+		{user: test.UserNoRoles, object: query, action: write, allow: false},
+		{user: test.UserNoRoles, object: query, action: run, allow: false},
+		{user: test.UserNoRoles, object: observerQuery, action: read, allow: true},
+		{user: test.UserNoRoles, object: observerQuery, action: write, allow: false},
+		{user: test.UserNoRoles, object: query, action: run, allow: false},
+
+		// Global observer can read
+		{user: test.UserObserver, object: query, action: read, allow: true},
+		{user: test.UserObserver, object: query, action: write, allow: false},
+		{user: test.UserObserver, object: query, action: run, allow: false},
+		{user: test.UserObserver, object: observerQuery, action: read, allow: true},
+		{user: test.UserObserver, object: observerQuery, action: write, allow: false},
+		// Can run observer query
+		{user: test.UserObserver, object: observerQuery, action: run, allow: true},
+
+		// Global maintainer can read/write/run
+		{user: test.UserMaintainer, object: query, action: read, allow: true},
+		{user: test.UserMaintainer, object: query, action: write, allow: true},
+		{user: test.UserMaintainer, object: query, action: run, allow: true},
+		{user: test.UserMaintainer, object: observerQuery, action: read, allow: true},
+		{user: test.UserMaintainer, object: observerQuery, action: write, allow: true},
+		{user: test.UserMaintainer, object: observerQuery, action: run, allow: true},
+
+		// Global admin can read/write
+		{user: test.UserAdmin, object: query, action: read, allow: true},
+		{user: test.UserAdmin, object: query, action: write, allow: true},
+		{user: test.UserAdmin, object: query, action: run, allow: true},
+		{user: test.UserAdmin, object: observerQuery, action: read, allow: true},
+		{user: test.UserAdmin, object: observerQuery, action: write, allow: true},
+		{user: test.UserAdmin, object: observerQuery, action: run, allow: true},
+
+		// Team observer read
+		{user: teamObserver, object: query, action: read, allow: true},
+		{user: teamObserver, object: query, action: write, allow: false},
+		{user: teamObserver, object: query, action: run, allow: false},
+		{user: teamObserver, object: observerQuery, action: read, allow: true},
+		{user: teamObserver, object: observerQuery, action: write, allow: false},
+		// Can run observer query
+		{user: teamObserver, object: observerQuery, action: run, allow: true},
+
+		// Team maintainer can read/write
+		{user: teamMaintainer, object: query, action: read, allow: true},
+		{user: teamMaintainer, object: query, action: write, allow: true},
```

Comment:
Team maintainers no longer have the ability to create/save queries. Team maintainers can only write (not save), and run, a custom query.

Does this mean that Team maintainer should not have write access?